llvm/lib/Object/COFFImportFile.cpp

```diff
@@ -729,7 +729,10 @@ Error writeImportLibrary(StringRef ImportName, StringRef Path,
         else if (Machine == IMAGE_FILE_MACHINE_I386 &&
                  applyNameType(IMPORT_NAME_NOPREFIX, Name) == E.ImportName)
           NameType = IMPORT_NAME_NOPREFIX;
-        else if (Name == E.ImportName)
+        else if (isArm64EC(M)) {
+          NameType = IMPORT_NAME_EXPORTAS;
```

Comment:
Btw, entirely unrelated side note, but I was wondering, do we have any actual evidence pointing towards the fact that this name type really should be called `EXPORTAS`? I guess the only official "upstream" reference to the name is that the MSVC tools support an `EXPORTAS` keyword in def files, to achieve this, but that doesn't strictly say that the name type should be called that.

My main gripe is that when dealing with import libraries, having _import_ entries be called `exportas`, feels a bit inconsitent. But I don't have any better suggestion, and using an upstream name probably still is better than making up a name of our own? (Or are the existing names, e.g. `IMPORT_NAME_UNDECORATE`, also made up locally?)

Reply:
While it's not documented, there is `IMPORT_OBJECT_NAME_EXPORTAS` defined in a public `winnt.h` header (and other `ImportNameType` otherwise follow `winnt.h` names). IIRC dumpbin also used that name.

It's not great when dealing with imports, but it seems better for consistency.

Reply:
Oh, ok - in that case the name does seem to be quite official, thanks for the reference!